ksqldb-engine/src/test/java/io/confluent/ksql/execution/ExpressionEvaluatorParityTest.java

```diff
@@ -160,7 +160,8 @@ public void shouldDereference() throws Exception {
     assertOrders("ITEMINFO->NAME", ITEM_NAME);
     assertOrders("ITEMINFO->CATEGORY->NAME", CATEGORY_NAME);
     assertOrders("'a-' + ITEMINFO->CATEGORY->NAME + '-b'", "a-cat-b");
-    assertOrdersError("ADDRESS->STREET + 'foo'", evalLogger(null));
+    // Due to https://github.com/confluentinc/ksql/issues/9136 the next assert cannot be updated.
+    // assertOrders("ADDRESS->STREET + 'foo'", null);
```

Comment:
@suhas-satish As a sanity check, while addressing the null handling issues, I think I turned up a new, subtle issue.  

My solution is to tag that new issue and merge this PR.  Lemme know if that sounds reasonable to you.  

The issue was found via CI failing; if CI passes and you don't have concerns, I'll merge this PR later today!

Reply:
sounds good , thanks Jim